src/table/hooks/use-column-widths.ts

```diff
@@ -93,7 +93,7 @@ const useColumnWidths = (
 
   const [columnWidths, setColumnWidths] = useState(() => getColumnWidths(columns, tableWidth, getHugWidth));
 
-  useDidUpdateEffect(() => {
+  useOnPropsChange(() => {
```

Comment:
Using `useDidUpdateEffect` caused an flow of updates that triggered 3 renders each time the layout changed. The render cycles would look something like this
1. Stale layout and columnWidths is provied by`useSelectionContext`
2. The new layout is provided by `useSelectionContext` but the columnWidths remain stale
3. The new columnWidths is provided by `useSelectionContext`

By not wrapping the adjusting the state in a useEffect, the step 3 is removed.